framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2685,6 +2685,15 @@ private AnnotatedDeclaredType getFunctionalInterfaceType(Tree lambdaTree,
                 assertFunctionalInterface(javacTypes, (Type) trees.getTypeMirror(getPath(cast.getType())), parentTree, lambdaTree);
                 return (AnnotatedDeclaredType)getAnnotatedType(cast.getType());
 
+            case NEW_CLASS: {
```

Comment:
Good fix!  One really nitpicky point is that the surrounding blocks don't use the braces, {}.  But, if no one has an objection I don't mind merging this and fixing the formatting. 

Thanks for the fix!